core/src/com/cloud/agent/resource/virtualnetwork/facade/AbstractConfigItemFacade.java

```diff
@@ -104,13 +109,25 @@ public static AbstractConfigItemFacade getInstance(final Class<? extends Network
         return instance;
     }
 
+
+    private static String appendUuidToJsonFiles(final String filename) {
+        String remoteFileName = new String(filename);
```

Comment:
Use of ``new String()`` is a significant performance problem as it bypasses the constant pool -- placing additional pressure on the heap and garbage collector.  The ``StringBuilder`` class is provided for these types of circumstances and avoids these performance problems.